import sys
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWebEngineWidgets import QWebEngineView
from PyQt5.QtCore import QUrl
import os


class Ui_Form(object):
    def setupUi(self, Form):
        Form.setObjectName("Form")
        Form.resize(749, 492)
        self.MachineTable = QtWidgets.QTableWidget(Form)
        self.MachineTable.setGeometry(QtCore.QRect(10, 10, 291, 201))
        self.MachineTable.setObjectName("MachineTable")
        self.MachineTable.setColumnCount(5)
        self.MachineTable.setRowCount(9)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(3, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(4, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(5, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(6, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(7, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setVerticalHeaderItem(8, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setHorizontalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setHorizontalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setHorizontalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setHorizontalHeaderItem(3, item)
        item = QtWidgets.QTableWidgetItem()
        self.MachineTable.setHorizontalHeaderItem(4, item)
        self.DetMachineTable = QtWidgets.QTableWidget(Form)
        self.DetMachineTable.setGeometry(QtCore.QRect(440, 10, 301, 201))
        self.DetMachineTable.setObjectName("DetMachineTable")
        self.DetMachineTable.setColumnCount(5)
        self.DetMachineTable.setRowCount(10)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(3, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(4, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(5, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(6, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(7, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(8, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setVerticalHeaderItem(9, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setHorizontalHeaderItem(0, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setHorizontalHeaderItem(1, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setHorizontalHeaderItem(2, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setHorizontalHeaderItem(3, item)
        item = QtWidgets.QTableWidgetItem()
        self.DetMachineTable.setHorizontalHeaderItem(4, item)
        self.CreateB = QtWidgets.QPushButton(Form)
        self.CreateB.setGeometry(QtCore.QRect(310, 10, 121, 61))
        self.CreateB.setObjectName("CreateB")
        self.VisB = QtWidgets.QPushButton(Form)
        self.VisB.setGeometry(QtCore.QRect(310, 170, 121, 41))
        self.VisB.setObjectName("VisB")
        self.Viewer = QWebEngineView(Form)
        self.Viewer.setGeometry(QtCore.QRect(10, 220, 731, 271))
        self.Viewer.setObjectName("Viewer")
        self.comboBox = QtWidgets.QComboBox(Form)
        self.comboBox.setGeometry(QtCore.QRect(310, 90, 121, 51))
        self.comboBox.setObjectName("comboBox")
        self.comboBox.addItem("")
        self.comboBox.addItem("")

        self.retranslateUi(Form)
        QtCore.QMetaObject.connectSlotsByName(Form)

    def retranslateUi(self, Form):
        _translate = QtCore.QCoreApplication.translate
        Form.setWindowTitle(_translate("Form", "Form"))
        item = self.MachineTable.verticalHeaderItem(0)
        item.setText(_translate("Form", "S0"))
        item = self.MachineTable.verticalHeaderItem(1)
        item.setText(_translate("Form", "S1"))
        item = self.MachineTable.verticalHeaderItem(2)
        item.setText(_translate("Form", "S2"))
        item = self.MachineTable.verticalHeaderItem(3)
        item.setText(_translate("Form", "S3"))
        item = self.MachineTable.verticalHeaderItem(4)
        item.setText(_translate("Form", "Новая строка"))
        item = self.MachineTable.verticalHeaderItem(5)
        item.setText(_translate("Form", "Новая строка"))
        item = self.MachineTable.verticalHeaderItem(6)
        item.setText(_translate("Form", "Новая строка"))
        item = self.MachineTable.verticalHeaderItem(7)
        item.setText(_translate("Form", "Новая строка"))
        item = self.MachineTable.verticalHeaderItem(8)
        item.setText(_translate("Form", "Новая строка"))
        item = self.MachineTable.horizontalHeaderItem(0)
        item.setText(_translate("Form", "0"))
        item = self.MachineTable.horizontalHeaderItem(1)
        item.setText(_translate("Form", "1"))
        item = self.MachineTable.horizontalHeaderItem(2)
        item.setText(_translate("Form", "Новый столбец"))
        item = self.MachineTable.horizontalHeaderItem(3)
        item.setText(_translate("Form", "Новый столбец"))
        item = self.MachineTable.horizontalHeaderItem(4)
        item.setText(_translate("Form", "Новый столбец"))
        item = self.DetMachineTable.verticalHeaderItem(0)
        item.setText(_translate("Form", "p0"))
        item = self.DetMachineTable.verticalHeaderItem(1)
        item.setText(_translate("Form", "p1"))
        item = self.DetMachineTable.verticalHeaderItem(2)
        item.setText(_translate("Form", "p2"))
        item = self.DetMachineTable.verticalHeaderItem(3)
        item.setText(_translate("Form", "p3"))
        item = self.DetMachineTable.verticalHeaderItem(4)
        item.setText(_translate("Form", "p4"))
        item = self.DetMachineTable.verticalHeaderItem(5)
        item.setText(_translate("Form", "Новая строка"))
        item = self.DetMachineTable.verticalHeaderItem(6)
        item.setText(_translate("Form", "Новая строка"))
        item = self.DetMachineTable.verticalHeaderItem(7)
        item.setText(_translate("Form", "Новая строка"))
        item = self.DetMachineTable.verticalHeaderItem(8)
        item.setText(_translate("Form", "Новая строка"))
        item = self.DetMachineTable.verticalHeaderItem(9)
        item.setText(_translate("Form", "Новая строка"))
        item = self.DetMachineTable.horizontalHeaderItem(0)
        item.setText(_translate("Form", "0"))
        item = self.DetMachineTable.horizontalHeaderItem(1)
        item.setText(_translate("Form", "1"))
        item = self.DetMachineTable.horizontalHeaderItem(2)
        item.setText(_translate("Form", "Новый столбец"))
        item = self.DetMachineTable.horizontalHeaderItem(3)
        item.setText(_translate("Form", "Новый столбец"))
        item = self.DetMachineTable.horizontalHeaderItem(4)
        item.setText(_translate("Form", "Новый столбец"))
        self.CreateB.setText(_translate("Form", "Построить \n"
" детерминированный \n"
" автомат"))
        self.VisB.setText(_translate("Form", "Визуализировать"))
        self.comboBox.setItemText(0, _translate("Form", "Исходный"))
        self.comboBox.setItemText(1, _translate("Form", "Детерминированный"))

class MainWindow(QtWidgets.QMainWindow):
    def __init__(self):
        super(MainWindow, self).__init__()
        self.ui = Ui_Form()
        self.ui.setupUi(self)
        self.ui.VisB.clicked.connect(self.visualize)
        with open('example.html', 'r') as f:
            html = f.read()


    def visualize(self):
        local_url = QUrl.fromLocalFile('os.path.abspath(os.path.join(os.path.dirname(__file__), "example.html"))')
        self.ui.Viewer.load(local_url)
        self.ui.Viewer.show()






app = QtWidgets.QApplication(sys.argv)
mainWindow = MainWindow()
widget = QtWidgets.QStackedWidget()
widget.setWindowTitle("Lab#4")

widget.addWidget(mainWindow)
widget.setFixedHeight(800)
widget.setFixedWidth(750)
widget.show()

sys.exit(app.exec_())